src/Nethermind/Nethermind.Evm/CodeAnalysis/ICodeInfo.cs

```diff
@@ -0,0 +1,18 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Evm.Precompiles;
+
+namespace Nethermind.Evm.CodeAnalysis;
+
+public interface ICodeInfo
+{
+    byte[] MachineCode { get; }
+    IPrecompile? Precompile { get; }
+    bool IsPrecompile => Precompile is not null;
+    ReadOnlyMemory<byte> TypeSection => Memory<byte>.Empty;
+    ReadOnlyMemory<byte> CodeSection => MachineCode;
+    ReadOnlyMemory<byte> DataSection => Memory<byte>.Empty;
+    bool ValidateJump(int destination, bool isSubroutine);
+}
```

Comment:
BTW this might get refactored more, but leave it at that for now as there are too many PR's to track at the moment.